aten/src/ATen/native/quantized/cpu/qnnpack_utils.h

```diff
@@ -84,3 +89,58 @@ inline std::pair<uint8_t, uint8_t> activationLimits(
   }
 }
 #endif
+
+namespace {
+std::vector<float> generate_requantization_scales(
+    const at::Tensor& weight_scales,
+    const float input_scale,
+    const float output_scale) {
+  auto num_output_channels = weight_scales.numel();
+  float* weight_scales_data = weight_scales.data_ptr<float>();
+  std::vector<float> requant_scales(num_output_channels, 1.f);
+  for (int i = 0; i < num_output_channels; ++i) {
+    requant_scales[i] = weight_scales_data[i] * input_scale / output_scale;
+  }
+  return requant_scales;
+}
+
+std::pair<at::Tensor, at::Tensor> make_zero_points_and_scales_tensor(
+    const at::Tensor& weight_contig
+    ) {
+  auto num_output_channels = weight_contig.size(0);
+  const auto qtype = weight_contig.qscheme();
+  at::Tensor weight_zp =
+    at::native::empty_cpu({1}, at::device(at::kCPU).dtype(at::kQUInt8));
+  // Adjust weight zero point, similar to weight data.
+  uint8_t* weight_zp_data;
+  if (qtype == at::kPerTensorAffine) {
+    weight_zp_data = (uint8_t*)weight_zp.data_ptr<c10::quint8>();
+    weight_zp_data[0] = (uint8_t)(weight_contig.q_zero_point() + 128);
```

Comment:
`q_zero_point()` returns int64_t. Why are we directly casting it to unit8_t here?

Reply:
Are we expecting zero point to be outside of uint8_t range after the addition of 128? My understanding was the it is already representable in 8 bits. I would assume you would want to go via 64->32->8 for rounding reason but I dont know what difference it makes here.